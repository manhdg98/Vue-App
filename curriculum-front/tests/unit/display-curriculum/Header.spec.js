import { mount, shallowMount } from '@vue/test-utils'
import Header from '@/components/display-curriculum/Header.vue'

describe('display-curriculum/Header.vue', () => {
  test('if snapshot is correct', () => {
    const wrapper = mount(Header, {
      propsData: {
        editField: '', // name, goal, or description
        selectedCurriculum: { name: 'Learn Vue.js' },
        toggleEdit: () => {},
        saveEdit: () => {},
        cancelEdit: () => {}
      }
    })
    expect(wrapper).toMatchSnapshot()
  })

  test('that title edit mode can display properly', () => {
    const wrapper = shallowMount(Header, {
      propsData: {
        editField: 'name',
        selectedCurriculum: { name: 'Learn Vue.js' },
        toggleEdit: () => {},
        saveEdit: () => {},
        cancelEdit: () => {}
      }
    })

    const input = wrapper.find('[data-test="name-edit-field"]')
    expect(input.exists()).toBe(true)
  })
})
