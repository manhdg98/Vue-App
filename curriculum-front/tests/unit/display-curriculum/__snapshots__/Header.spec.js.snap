// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`display-curriculum/Header.vue if snapshot is correct 1`] = `
<div class="row">
  <div class="col">
    <h1 class="hover-icon-container">
      Learn Vue.js
      <i role="button" class="v-icon notranslate togglable-icon v-icon--link mdi mdi-pencil-box-outline theme--light gray--text text--lighten-1"></i></h1>
    <div><button type="button" class="v-btn v-btn--flat v-btn--text theme--light v-size--small info--text"><span class="v-btn__content">Add Goal</span></button></div>
    <p class="hover-icon-container">

      <i role="button" class="v-icon notranslate togglable-icon v-icon--link mdi mdi-pencil-box-outline theme--light gray--text text--lighten-1" style="font-size: 16px;"></i></p>
  </div>
</div>
`;
